File: app/src/main/java/com/vkbot/manager/botbrain/SearchResult.java
package com.vkbot.manager.botbrain;

import java.io.Serial;
import java.io.Serializable;
import java.util.List;

/**
 * Класс для хранения результатов поиска ответа (Java 21 Record).
 * Содержит найденный элемент базы и список захваченных групп (из Regex или масок).
 * Версия 2.2 - Record Optimized.
 */
@SuppressWarnings("unused")
public record SearchResult(
    AnswerElement answer,
    List<String> capturedGroups
) implements Serializable {
    
    @Serial
    private static final long serialVersionUID = 1L;

    public SearchResult {
        capturedGroups = capturedGroups != null ? List.copyOf(capturedGroups) : List.of();
    }

    /**
     * Конструктор для быстрого создания без групп.
     */
    public SearchResult(AnswerElement answer) {
        this(answer, List.of());
    }

    // Методы для обратной совместимости API
    public AnswerElement getAnswer() { return answer; }
    public List<String> getCapturedGroups() { return capturedGroups; }
    public boolean hasGroups() { return !capturedGroups.isEmpty(); }
}

File: app/src/main/java/com/vkbot/manager/botbrain/SearchResult.kt
package com.vkbot.manager.botbrain

import java.io.Serializable

/**
 * Класс для хранения результатов поиска ответа (Kotlin data class).
 * Содержит найденный элемент базы и список захваченных групп (из Regex или масок).
 */
data class SearchResult(
    val answer: AnswerElement,
    val capturedGroups: List<String> = emptyList()
) : Serializable {
    
    fun hasGroups(): Boolean = capturedGroups.isNotEmpty()

}
